### PR TITLE
Fixed some error notices

### PR DESCRIPTION
**Pull request by aschempp**

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/Model.php [hunk 1]
```diff
@@ -1072,7 +1072,7 @@ protected static function find(array $arrOptions)
 		}
 
 		$objStatement = static::preFind($objStatement);
-		$objResult = $objStatement->execute($arrOptions['value']);
+		$objResult = $objStatement->execute($arrOptions['value'] ?? null);
 
 		if ($objResult->numRows < 1)
```
Comment (bytehead) [inline core-bundle/src/Resources/contao/library/Contao/Model.php:1077]: if `$objResult` is null, access to an undefined property will raise a warning in PHP 8.

```suggestion
		if (null !== $objResult && $objResult->numRows < 1)
```

**Review by bytehead — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Model.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Resources/contao/library/Contao/Model.php:1077]: why would `$objStatement->execute()` return `null`? It's a database query, not a model collection 😉

Comment (bytehead) [reply core-bundle/src/Resources/contao/library/Contao/Model.php:1077]: Ops, I've missed the parenthesis 🙈 😂

**Review by bytehead — DISMISSED** [review]

File: core-bundle/src/Resources/contao/library/Contao/DcaExtractor.php [hunk 2]
```diff
@@ -421,7 +423,7 @@ protected function createExtract()
 						$table = substr($config['foreignKey'], 0, strrpos($config['foreignKey'], '.'));
 					}
 
-					$arrRelations[$field] = array_merge(array('table'=>$table, 'field'=>'id'), $config['relation']);
+					$arrRelations[$field] = array_merge(array('table'=>$table, 'field'=>'id', 'load'=>'lazy', 'type'=>null), $config['relation']);
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/DcaExtractor.php]: This should be fixed where the values are read and not where they are written.

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/library/Contao/DcaExtractor.php (hunk 2 above)
Comment (aschempp) [reply core-bundle/src/Resources/contao/library/Contao/DcaExtractor.php]: Shouldn't we make sure the relation data is always consistent?

Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/DcaExtractor.php]: No, because that is a behavior change and thus a possible BC break.

Comment (aschempp) [reply core-bundle/src/Resources/contao/library/Contao/DcaExtractor.php]: Changed in acc9110172ba8f3aa53526b159221c94013b0bb7

File: core-bundle/src/Resources/contao/library/Contao/Model/QueryBuilder.php [hunk 3]
```diff
@@ -45,9 +45,9 @@ public static function find(array $arrOptions)
 			foreach ($objBase->getRelations() as $strKey=>$arrConfig)
 			{
 				// Automatically join the single-relation records
-				if ($arrConfig['load'] == 'eager' || $arrOptions['eager'])
+				if (($arrConfig['load'] ?? null) == 'eager' || ($arrOptions['eager'] ?? false))
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/Model/QueryBuilder.php]:
```suggestion
				if (($arrConfig['load'] ?? null) == 'eager' || ($arrOptions['eager'] ?? null))
```

Using `?? false` is a behavior change again, even though I don't think it makes a difference. @ausi How would you handle this?

Comment (ausi) [reply core-bundle/src/Resources/contao/library/Contao/Model/QueryBuilder.php]: I would use `?? null` for consistency but it is exactly the same in this case I think.

Comment (m-vo) [reply core-bundle/src/Resources/contao/library/Contao/Model/QueryBuilder.php]: I remember this discussion :upside_down_face:

Comment (aschempp) [reply core-bundle/src/Resources/contao/library/Contao/Model/QueryBuilder.php]: Changed in 9cd2ac02efc6ba2e8b36a22be0659c01e32c870d

**Comment by leofeyer** [comment]
Please rebase the PR onto the `4.11` branch.

**Comment by aschempp** [comment]
All changes have already been applied (probably through other PRs) in 4.11 😆

**Pull request closed without merging**